--- client_server/CMakeLists.txt.orig	2024-12-24 03:34:58
+++ client_server/CMakeLists.txt	2024-12-24 07:24:17
@@ -31,7 +31,7 @@
 # websocket functionality) which was not stable before that version.
 # Actually, it is probably not stable before 1.75.0, but if we go that
 # high we cannot build on Ubuntu 20.04 anymore.
-find_package(Boost 1.71.0 COMPONENTS system program_options date_time filesystem REQUIRED)
+find_package(Boost 1.81.0 COMPONENTS system program_options date_time filesystem REQUIRED)
 
 # OpenSSL needs to be linked in explicitly (probably because the boost
 # material referring to it is all in headers?).
