--- lib/filetree.c.orig	2022-05-03 23:39:20.000000000 -0500
+++ lib/filetree.c	2022-05-06 08:24:49.000000000 -0500
@@ -741,7 +741,8 @@
 	size_t ns1_size = 0, ns2_size = 0;
 	const struct __xar_file_t * child1 = NULL, * child2 = NULL;
 	const uint keys_to_ignore_count = 1;
-	char * keys_to_ignore[keys_to_ignore_count] = { "id" }; // ID is allowed ot mismatch
+	char * keys_to_ignore[keys_to_ignore_count];
+	* keys_to_ignore[keys_to_ignore_count] = { "id" }; // ID is allowed ot mismatch
 	
 	// If the two pointers match, call it the same.
 	if (f1 == f2)  {
