--- CMakeLists.txt.orig	2015-09-11 10:30:41.000000000 -0500
+++ CMakeLists.txt	2018-01-25 07:39:36.000000000 -0600
@@ -6,7 +6,7 @@
 option(WITH_MONGO "Build with support for MongoDB" OFF)
 option(WITH_JSON "Build with support for JSON" ON)
 option(ENABLE_DEVELOPMENT "Enable development specific compiler and linker flags" OFF)
-option(RESPECT_INSTALL_PREFIX_CONFIG "Respect CMAKE_INSTALL_PREFIX for /etc" OFF)
+option(RESPECT_INSTALL_PREFIX_CONFIG "Respect CMAKE_INSTALL_PREFIX for /etc" ON)
 option(WITH_WERROR "Build with -Werror" OFF)
 option(WITH_PFRING "Build with PF_RING ZC for send (10 GigE)" OFF)
 
