--- src/Makefile.orig	2013-11-25 20:40:57.000000000 -0500
+++ src/Makefile	2013-11-25 20:40:46.000000000 -0500
@@ -25,7 +26,7 @@
 all : $(TARGETS) pgmres.oct
 
 $(GSVD_TARGET) : $(GSVD_DEPENDS) $(GSVD_OBJECTS)
-	$(MKOCTFILE) $(DEFINES) $(GSVD_OBJECTS) -o $@ ${LAPACK_LIBS}
+	$(MKOCTFILE) $(DEFINES) $(GSVD_OBJECTS) -o $@ ${LAPACK_LIBS} ${BLAS_LIBS}
 
 $(GSVD_TEST) : $(GSVD_TARGET)
 
@@ -46,7 +47,7 @@
 	$(MKOCTFILE) $(DEFINES) $< -o $@
 
 %.oct: %.cc
-	mkoctfile $(DEFINES) $< -o $@ ${LAPACK_LIBS}
+	mkoctfile $(DEFINES) $< -o $@ ${LAPACK_LIBS} ${BLAS_LIBS}
 
 .phony: test
 test: $(GSVD_TEST)
