--- aux/zeekctl/aux/pysubnettree/setup.py.orig	2020-02-21 20:42:15.000000000 -0600
+++ aux/zeekctl/aux/pysubnettree/setup.py	2020-02-21 20:55:39.000000000 -0600
@@ -19,6 +19,12 @@
 with open('README') as file:
     long_description = file.read()
 
+if sys.platform.startswith("darwin"):
+    # On recent macos versions (mojave) it is necessary to specify that libc++ is used instead of libstdc++.
+    # Furthermore, '-Wl,-undefined,dynamic_lookup' is necessary to link the right libraries.
+    libs += ["-stdlib=libc++", '-Wl,-undefined,dynamic_lookup']
+    extra_compile_args = ["-stdlib=libc++"]
+
 setup(name="pysubnettree",
     version="0.31", # Filled in automatically.
     maintainer="The Zeek Project",
@@ -33,4 +39,3 @@
             depends=["SubnetTree.h", "patricia.h"]),
         ]
 )
-
