--- libgui/src/main-window.cc.orig	2017-02-22 12:01:55.000000000 -0600
+++ libgui/src/main-window.cc	2017-10-02 10:24:08.000000000 -0500
@@ -562,8 +562,6 @@
   connect (worker_thread, SIGNAL (started (void)),
            reader, SLOT (process ()));
 
-  connect (reader, SIGNAL (finished (void)), worker_thread, SLOT (quit ()));
-
   connect (reader, SIGNAL (finished (void)), reader, SLOT (deleteLater ()));
 
   connect (worker_thread, SIGNAL (finished (void)),
@@ -2391,3 +2389,13 @@
 {
   _clipboard->clear (QClipboard::Clipboard);
 }
+
+void
+main_window::handle_octave_finished (int exit_status)
+  {
+    worker_thread->quit();
+    worker_thread->wait();
+ 
+    qApp->exit (exit_status);
+  }
+
