--- src/core/renderer/StelQGLInterleavedArrayVertexBufferBackend.cpp.orig	2014-10-11 06:24:38.000000000 -0500
+++ src/core/renderer/StelQGLInterleavedArrayVertexBufferBackend.cpp	2015-01-20 12:49:11.000000000 -0600
@@ -47,7 +47,7 @@
 
 StelQGLInterleavedArrayVertexBufferBackend::~StelQGLInterleavedArrayVertexBufferBackend()
 {
-	std::free(vertices);
+	::free(vertices);
 	if(NULL != projectedPositions)
 	{
 		::free(projectedPositions);
@@ -66,14 +66,14 @@
 		vertices = static_cast<char*>(::realloc(vertices, vertexCapacity * vertexStride));
 	}
 
-	std::memcpy(vertices + vertexCount * vertexStride, vertexInPtr, vertexBytes);
+	::memcpy(vertices + vertexCount * vertexStride, vertexInPtr, vertexBytes);
 	++vertexCount;
 }
 
 void StelQGLInterleavedArrayVertexBufferBackend::getVertex
 	(const int index, void* const vertexOutPtr) const
 {
-	std::memcpy(vertexOutPtr, vertices + index * vertexStride, vertexBytes);
+	::memcpy(vertexOutPtr, vertices + index * vertexStride, vertexBytes);
 }
 
 void StelQGLInterleavedArrayVertexBufferBackend::
