--- unsupported/Eigen/CMakeLists.txt.orig	2015-06-16 04:53:07.000000000 -0500
+++ unsupported/Eigen/CMakeLists.txt	2015-06-19 09:19:55.000000000 -0500
@@ -1,6 +1,6 @@
 set(Eigen_HEADERS AdolcForward AlignedVector3 ArpackSupport AutoDiff BVH FFT IterativeSolvers KroneckerProduct LevenbergMarquardt
                   MatrixFunctions MoreVectorization MPRealSupport NonLinearOptimization NumericalDiff OpenGLSupport Polynomials
-                  Skyline SparseExtra Splines
+                  SVD Skyline SparseExtra Splines
    )
 
 install(FILES
