diff --git a/docs/rdiff-backup-old.1 b/docs/rdiff-backup-old.1.template
similarity index 100%
rename from docs/rdiff-backup-old.1
rename to docs/rdiff-backup-old.1.template
diff --git docs/rdiff-backup-statistics.1 docs/rdiff-backup-statistics.1
deleted file mode 100644
index aa944d3e9..000000000
--- docs/rdiff-backup-statistics.1
+++ /dev/null
@@ -1,92 +0,0 @@
-.TH RDIFF-BACKUP-STATISTICS 1 "{{ month_year }}" "Version {{ version }}" "User Manuals" \" -*- nroff -*-
-.SH NAME
-rdiff-backup-statistics \- summarize rdiff-backup statistics files
-.SH SYNOPSIS
-.B rdiff-backup-statistics
-.BI [\-\-begin-time " time" ]
-.BI [\-\-end-time " time" ]
-.BI [\-\-minimum-ratio " ratio" ]
-.B [\-\-null-separator]
-.B [\-\-quiet]
-.B [-h|\-\-help]
-.B [-V|\-\-version]
-.I repository
-
-.SH DESCRIPTION
-.BI rdiff-backup-statistics
-reads the matching statistics files in a backup repository made by
-.B rdiff-backup
-and prints some summary statistics to the screen.  It does not alter
-the repository in any way.
-
-The required argument is the pathname of the root of an rdiff-backup
-repository.  For instance, if you ran "rdiff-backup in out", you could
-later run "rdiff-backup-statistics out".
-
-The output has two parts.  The first is simply an average of the all
-matching session_statistics files.  The meaning of these fields is
-explained in the FAQ included in the package, and also at
-.IR https://rdiff-backup.net/docs/FAQ.html#statistics .
-
-The second section lists some particularly significant files
-(including directories).  These files are either contain a lot of
-data, take up increment space, or contain a lot of changed files.  All
-the files that are above the minimum ratio (default 5%) will be
-listed.
-
-If a file or directory is listed, its contributions are subtracted
-from its parent.  That is why the percentage listed after a directory
-can be larger than the percentage of its parent.  Without this, the
-root directory would always be the largest, and the output would be
-boring.
-
-.SH OPTIONS
-.TP
-.BI \-\-begin-time " time"
-Do not read statistics files older than
-.IR time .
-By default, all statistics files will be read.
-.I time
-should be in the same format taken by \-\-restore-as-of.  (See
-.B TIME FORMATS
-in the rdiff-backup man page for details.)
-.TP
-.BI \-\-end-time " time"
-Like
-.B \-\-begin-time
-but exclude statistics files later than
-.IR time .
-.TP
-.B -h, \-\-help
-Output a short usage description and exit.
-.TP
-.BI \-\-minimum-ratio " ratio"
-Print all directories contributing more than the given ratio to the
-total.  The default value is .05, or 5 percent.
-.TP
-.B \-\-null-separator
-Specify that the lines of the file_statistics file are separated by
-nulls (\\0).  The default is to assume that newlines separate.  Use
-this switch if rdiff-backup was run with the \-\-null-separator when
-making the given repository.
-.TP
-.B \-\-quiet
-Suppress printing of the "Processing statistics from session..."
-output lines.
-.TP
-.B -V, \-\-version
-Output full path to command and version then exit.
-
-.SH BUGS
-When aggregating multiple statistics files, some directories above
-(but close to) the minimum ratio may not be displayed.  For this
-reason, you may want to set the minimum-ratio lower than need.
-
-.SH AUTHOR
-Ben Escoto <ben@emerose.org>, based on original script by Dean Gaudet.
-
-.SH SEE ALSO
-.BR rdiff-backup (1),
-.BR python (1).
-The rdiff-backup web page is at
-.IR https://rdiff-backup.net/ .
diff --git a/setup.py b/setup.py
index 92ef59e68..834eefd87 100755
--- setup.py
+++ setup.py
@@ -10,6 +10,7 @@
 # we need all this to extend the distutils/setuptools commands
 from setuptools import setup, Extension, Command
 import setuptools.command.build_py
+import setuptools.command.sdist
 from distutils.debug import DEBUG
 import distutils.command.clean
 from distutils import log
@@ -42,24 +43,25 @@
         if LFLAGS or LIBS:
             lflags_arg = LFLAGS + LIBS
 
-        if LIBRSYNC_DIR and len(sys.argv) > 1:  # should only be under Windows
+        if LIBRSYNC_DIR and len(sys.argv) > 1:
             incdir_list = [os.path.join(LIBRSYNC_DIR, "include")]
             libdir_list = [os.path.join(LIBRSYNC_DIR, "lib")]
-            rsyncdll_src = os.path.join(LIBRSYNC_DIR, "bin", "rsync.dll")
-            rsyncdll_dst = os.path.join("src", "rdiff_backup", "rsync.dll")
-            # rather ugly workaround, but it should be good enough
-            if "clean" in sys.argv:
-                if os.path.exists(rsyncdll_dst) and "--all" in sys.argv:
-                    print(f"removing {rsyncdll_dst}")
-                    if "--dry-run" not in sys.argv:
-                        os.remove(rsyncdll_dst)
-            elif ("--version" not in sys.argv and "-V" not in sys.argv
-                  and "--help" not in sys.argv):
-                if (not os.path.exists(rsyncdll_dst)
-                        or not filecmp.cmp(rsyncdll_src, rsyncdll_dst)):
-                    print(f"copying {rsyncdll_src} -> {rsyncdll_dst}")
-                    if "--dry-run" not in sys.argv:
-                        shutil.copyfile(rsyncdll_src, rsyncdll_dst)
+            if os.name == "nt":
+                rsyncdll_src = os.path.join(LIBRSYNC_DIR, "bin", "rsync.dll")
+                rsyncdll_dst = os.path.join("src", "rdiff_backup", "rsync.dll")
+                # rather ugly workaround, but it should be good enough
+                if "clean" in sys.argv:
+                    if os.path.exists(rsyncdll_dst) and "--all" in sys.argv:
+                        print(f"removing {rsyncdll_dst}")
+                        if "--dry-run" not in sys.argv:
+                            os.remove(rsyncdll_dst)
+                elif ("--version" not in sys.argv and "-V" not in sys.argv
+                      and "--help" not in sys.argv):
+                    if (not os.path.exists(rsyncdll_dst)
+                            or not filecmp.cmp(rsyncdll_src, rsyncdll_dst)):
+                        print(f"copying {rsyncdll_src} -> {rsyncdll_dst}")
+                        if "--dry-run" not in sys.argv:
+                            shutil.copyfile(rsyncdll_src, rsyncdll_dst)
         if "-lrsync" in LIBS:
             libname = []
 
@@ -71,7 +73,7 @@
 # --- extend the build command to execute a command ---
 
 
-class build_exec(Command):
+class pre_build_exec(Command):
     description = 'build template files executing a shell command'
     user_options = [
         # The format is (long option, short option, description).
@@ -120,7 +122,7 @@ def run(self):
 # --- extend the build command to do templating of files ---
 
 
-class build_templates(Command):
+class pre_build_templates(Command):
     description = 'build template files replacing {{ }} placeholders'
     user_options = [
         # The format is (long option, short option, description).
@@ -171,14 +173,23 @@ def run(self):
 
 
 class build_py(setuptools.command.build_py.build_py):
-    """Inject our build sub-command in the build step"""
+    """Inject our build sub-command in the build_py step"""
 
     def run(self):
-        self.run_command('build_exec')
-        self.run_command('build_templates')
+        self.run_command('pre_build_exec')
+        self.run_command('pre_build_templates')
         setuptools.command.build_py.build_py.run(self)
 
 
+class sdist(setuptools.command.sdist.sdist):
+    """Inject our build sub-command in the sdist step"""
+
+    def run(self):
+        self.run_command('pre_build_exec')
+        self.run_command('pre_build_templates')
+        setuptools.command.sdist.sdist.run(self)
+
+
 # --- extend the clean command to remove templated and exec files ---
 
 class clean(distutils.command.clean.clean):
@@ -191,10 +202,10 @@ def initialize_options(self):
 
     def finalize_options(self):
         """Post-process options."""
-        # take over the option from our build_templates command
-        self.set_undefined_options('build_templates',
+        # take over the option from our pre_build_templates command
+        self.set_undefined_options('pre_build_templates',
                                    ('template_files', 'template_files'))
-        self.set_undefined_options('build_exec',
+        self.set_undefined_options('pre_build_exec',
                                    ('commands', 'commands'))
         super().finalize_options()
 
@@ -283,10 +294,10 @@ def run(self):
         ),
     ],
     data_files=[
-        ("share/man/man1", ["build/rdiff-backup.1",
-                            "build/rdiff-backup-old.1",
-                            "build/rdiff-backup-delete.1",
-                            "build/rdiff-backup-statistics.1"]),
+        ("share/man/man1", ["dist/rdiff-backup.1",
+                            "dist/rdiff-backup-old.1",
+                            "dist/rdiff-backup-delete.1",
+                            "dist/rdiff-backup-statistics.1"]),
         (
             "share/doc/rdiff-backup", [
                 "CHANGELOG.adoc",
@@ -306,29 +317,30 @@ def run(self):
     # options is a hash of hash with command -> option -> value
     # the value happens here to be a list of file couples/tuples
     options={
-        'build_templates': {'template_files': [
+        'pre_build_templates': {'template_files': [
             ("tools/rdiff-backup.spec.template", "build/rdiff-backup.spec"),
             ("tools/rdiff-backup.spec.template-fedora", "build/rdiff-backup.fedora.spec"),
-            ("docs/rdiff-backup-old.1", "build/rdiff-backup-old.1"),
+            ("docs/rdiff-backup-old.1.template", "dist/rdiff-backup-old.1"),
         ]},
-        "build_exec": {"commands": [
+        "pre_build_exec": {"commands": [
             ("asciidoctor -b manpage -a revdate=\"{date}\" "
              "-a revnumber=\"{ver}\" -o {outfile} {infile}",
-             ("docs", "rdiff-backup.1.adoc"), ("build", "rdiff-backup.1")),
+             ("docs", "rdiff-backup.1.adoc"), ("dist", "rdiff-backup.1")),
             ("asciidoctor -b manpage -a revdate=\"{date}\" "
              "-a revnumber=\"{ver}\" -o {outfile} {infile}",
              ("docs", "rdiff-backup-statistics.1.adoc"),
-             ("build", "rdiff-backup-statistics.1")),
+             ("dist", "rdiff-backup-statistics.1")),
             ("asciidoctor -b manpage -a revdate=\"{date}\" "
              "-a revnumber=\"{ver}\" -o {outfile} {infile}",
              ("docs", "rdiff-backup-delete.1.adoc"),
-             ("build", "rdiff-backup-delete.1")),
+             ("dist", "rdiff-backup-delete.1")),
         ]},
     },
     cmdclass={
-        'build_exec': build_exec,
-        'build_templates': build_templates,
+        'pre_build_exec': pre_build_exec,
+        'pre_build_templates': pre_build_templates,
         'build_py': build_py,
+        'sdist': sdist,
         'clean': clean,
     },
     install_requires=[
