--- Makefile.in.orig	2025-05-29 09:35:12
+++ Makefile.in	2025-05-30 08:05:17
@@ -355,10 +355,15 @@
 @endif
 install-test: install-core
 	@echo "Post-install test of [package require $(tx.name.pkg) $(tx.version)]..."; \
+	set xtra=""; \
+	if [ x != "x$(DESTDIR)" ]; then \
+		xtra='set ::auto_path [linsert $$::auto_path 0 [file normalize $(DESTDIR)$(TCLLIBDIR)/..]];'; \
+	fi; \
 	if echo \
-		'set c 0; ' \
+		'set c 0; ' $$xtra \
 		'@TEAISH_POSTINST_PREREQUIRE@' \
-		'if {[catch {package require $(tx.name.pkg) $(tx.version)}]} {incr c};' \
+		'if {[catch {package require $(tx.name.pkg) $(tx.version)} xc]} {incr c};' \
+		'if {$$c && "" ne $$xc} {puts $$xc; puts "auto_path=$$::auto_path"};' \
 		'exit $$c' \
 		| $(TCLSH) ; then \
 		echo "passed"; \
