--- Makefile.in.orig	2025-02-06 07:59:25
+++ Makefile.in	2025-02-06 15:59:15
@@ -127,11 +127,11 @@
 	$(CC) -c sqlite3.c -o $@ $(CFLAGS) $(CFLAGS.libsqlite3)
 
 libsqlite3.LIB = libsqlite3$(T.lib)
-libsqlite3.SO = libsqlite3$(T.dll)
+libsqlite3.SO = libsqlite3
 
 $(libsqlite3.SO): sqlite3.o
 	$(CC) -o $@  sqlite3.o $(LDFLAGS.shlib) \
-		$(LDFLAGS) $(LDFLAGS.libsqlite3) $(LDFLAGS.libsqlite3.soname)
+		$(LDFLAGS) $(LDFLAGS.libsqlite3) -current_version 9.7.0 -compatibility_version 9.0.0 $(LDFLAGS.libsqlite3.soname)
 all: $(libsqlite3.SO)
 
 $(libsqlite3.LIB): sqlite3.o
@@ -143,10 +143,10 @@
 	@echo "Setting up $(libsqlite3.SO) symlinks..."; \
 		cd "$(install-dir.lib)" || exit $$?; \
 		rm -f $(libsqlite3.SO).0 $(libsqlite3.SO).$(PACKAGE_VERSION) || exit $$?; \
-		mv $(libsqlite3.SO) $(libsqlite3.SO).$(PACKAGE_VERSION) || exit $$?; \
-		ln -s $(libsqlite3.SO).$(PACKAGE_VERSION) $(libsqlite3.SO) || exit $$?; \
-		ln -s $(libsqlite3.SO).$(PACKAGE_VERSION) $(libsqlite3.SO).0 || exit $$?; \
-		ls -la $(libsqlite3.SO) $(libsqlite3.SO).[03]*; \
+		mv $(libsqlite3.SO) $(libsqlite3.SO).$(PACKAGE_VERSION)$(T.dll) || exit $$?; \
+		ln -s $(libsqlite3.SO).$(PACKAGE_VERSION)$(T.dll) $(libsqlite3.SO)$(T.dll) || exit $$?; \
+		ln -s $(libsqlite3.SO).$(PACKAGE_VERSION)$(T.dll) $(libsqlite3.SO).0$(T.dll) || exit $$?; \
+		ls -la $(libsqlite3.SO)$(T.dll) $(libsqlite3.SO).[03]*$(T.dll); \
 		if [ -e $(libsqlite3.SO).0.8.6 ]; then \
 			echo "ACHTUNG: legacy libtool-compatible install found. Re-linking it..."; \
 			rm -f libsqlite3.la $(libsqlite3.SO).0.8.6 || exit $$?; \
