diff --git a/src/Makefile.am b/src/Makefile.am
index 694c0a35..f76c8fb3 100644
--- src/Makefile.am
+++ src/Makefile.am
@@ -25,6 +25,7 @@ fsearch_NOINST_H_FILES = fsearch.h \
 			 fsearch_thread_pool.h \
 			 fsearch_window.h \
 			 fsearch_window_actions.h \
+			 fsearch_limits.h \
 			 iconstore.h \
 			 list_model.h \
 			 listview.h \
diff --git a/src/fsearch.c b/src/fsearch.c
index 112787e3..457e363b 100644
--- src/fsearch.c
+++ src/fsearch.c
@@ -24,7 +24,6 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <limits.h>
-#include <linux/limits.h>
 #include <string.h>
 #include <stdint.h>
 #include <glib.h>
@@ -32,6 +31,7 @@
 #include <glib/gi18n.h>
 #include "fsearch.h"
 #include "fsearch_config.h"
+#include "fsearch_limits.h"
 #include "list_model.h"
 #include "resources.h"
 #include "clipboard.h"
diff --git a/src/fsearch_config.c b/src/fsearch_config.c
index 42ea69d0..1e8cdb3e 100644
--- src/fsearch_config.c
+++ src/fsearch_config.c
@@ -21,9 +21,9 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <linux/limits.h>
 
 #include "fsearch_config.h"
+#include "fsearch_limits.h"
 #include "debug.h"
 
 const char *config_file_name = "fsearch.conf";
diff --git a/src/fsearch_limits.h b/src/fsearch_limits.h
new file mode 100644
index 00000000..b670ee23
--- /dev/null
+++ src/fsearch_limits.h
@@ -0,0 +1,26 @@
+/*
+   FSearch - A fast file search utility
+   Copyright © 2016 Christian Boxdörfer
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, see <http://www.gnu.org/licenses/>.
+   */
+
+#pragma once
+
+#include <limits.h>
+
+#ifndef PATH_MAX
+#define	PATH_MAX 4096 /* max # of characters in a path name */
+#endif
+
diff --git a/src/fsearch_window.c b/src/fsearch_window.c
index 6f17d5be..fece5ba4 100644
--- src/fsearch_window.c
+++ src/fsearch_window.c
@@ -20,7 +20,6 @@
 #include <config.h>
 #endif
 
-#include <linux/limits.h>
 #include <glib/gi18n.h>
 #include "fsearch_window.h"
 #include "fsearch_window_actions.h"
@@ -33,6 +32,7 @@
 #include "database_search.h"
 #include "listview.h"
 #include "debug.h"
+#include "fsearch_limits.h"
 
 struct _FsearchApplicationWindow {
     GtkApplicationWindow parent_instance;
diff --git a/src/fsearch_window_actions.c b/src/fsearch_window_actions.c
index 787d218e..1d17eb5a 100644
--- src/fsearch_window_actions.c
+++ src/fsearch_window_actions.c
@@ -20,12 +20,11 @@
 #include <config.h>
 #endif
 
-#include <limits.h>
-#include <linux/limits.h>
 #include <gio/gdesktopappinfo.h>
 #include <glib/gi18n.h>
 
 #include "fsearch_window_actions.h"
+#include "fsearch_limits.h"
 #include "clipboard.h"
 #include "database_search.h"
 #include "utils.h"
diff --git a/src/utils.c b/src/utils.c
index 64199264..5068d902 100644
