--- epan/wslua/lrexlib_glib_f.c.orig	2015-08-12 11:47:43.000000000 -0500
+++ epan/wslua/lrexlib_glib_f.c	2015-10-04 07:10:10.000000000 -0500
@@ -26,8 +26,8 @@
 */
 
 #include <glib.h>
-#include "lua.h"
-#include "lauxlib.h"
+#include <lua-5.2/lua.h>
+#include <lua-5.2/lauxlib.h>
 #include "lrexlib.h"
 
 #define VERSION_GLIB (GLIB_MAJOR_VERSION*100 + GLIB_MINOR_VERSION)
 
--- epan/wslua/lrexlib_glib.c.orig	2015-08-12 11:47:43.000000000 -0500
+++ epan/wslua/lrexlib_glib.c	2015-10-04 07:10:31.000000000 -0500
@@ -51,8 +51,8 @@
 #include <ctype.h>
 #include <glib.h>
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <lua-5.2/lua.h>
+#include <lua-5.2/lauxlib.h>
 #include "lrexlib.h"
 
 extern int Gregex_get_flags (lua_State *L);
 
--- epan/wslua/lrexlib.c.orig	2015-08-12 11:47:42.000000000 -0500
+++ epan/wslua/lrexlib.c	2015-10-04 07:11:10.000000000 -0500
@@ -39,8 +39,8 @@
 #include <string.h>
 #include <glib.h>
 
-#include "lua.h"
-#include "lauxlib.h"
+#include <lua-5.2/lua.h>
+#include <lua-5.2/lauxlib.h>
 #include "lrexlib.h"
 
 #define N_ALIGN sizeof(int)
 
--- epan/wslua/lua_bitop.c.orig	2015-08-12 11:47:43.000000000 -0500
+++ epan/wslua/lua_bitop.c	2015-10-03 13:50:48.000000000 -0500
@@ -29,8 +29,8 @@
 #define LUA_BITOP_VERSION	"1.0.2"
 
 #define LUA_LIB
-#include <lua.h>
-#include <lauxlib.h>
+#include <lua-5.2/lua.h>
+#include <lua-5.2/lauxlib.h>
 
 #include "lua_bitop.h"
 
 