--- configure.orig	2015-10-30 19:00:38.000000000 -0500
+++ configure	2015-10-30 19:09:49.000000000 -0500
@@ -33970,11 +33970,11 @@
 		# searches the specified directory.
 		#
 		wireshark_save_CPPFLAGS="$CPPFLAGS"
-		CPPFLAGS="$CPPFLAGS -I$lua_dir/include"
+		CPPFLAGS="$CPPFLAGS -I$lua_dir/include/lua-5.2"
 		wireshark_save_LIBS="$LIBS"
-		LIBS="$LIBS -L$lua_dir/lib -llua -lm"
+		LIBS="$LIBS -L$lua_dir/lib/lua-5.2 -llua -lm"
 		wireshark_save_LDFLAGS="$LDFLAGS"
-		LDFLAGS="$LDFLAGS -L$lua_dir/lib"
+		LDFLAGS="$LDFLAGS -L$lua_dir/lib/lua-5.2"
 
 		#
 		# Determine Lua version by reading the LUA_VERSION_NUM definition
@@ -33983,7 +33983,7 @@
 		#
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking Lua version" >&5
 $as_echo_n "checking Lua version... " >&6; }
-		[ -d "$lua_dir/include" ] && grep -rq 'LUA_VERSION_NUM.*501' "$lua_dir/include" && lua_ver=5.1
+		[ -d "$lua_dir/include/lua-5.2" ] && grep -rq 'LUA_VERSION_NUM.*501' "$lua_dir/include/lua-5.2" && lua_ver=5.1
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Lua ${lua_ver}" >&5
 $as_echo "Lua ${lua_ver}" >&6; }
 	else
@@ -34032,7 +34032,7 @@
 
 			if test "x$lua_dir" != "x"
 			then
-				LUA_INCLUDES="-I$lua_dir/include/lua${lua_ver}"
+				LUA_INCLUDES="-I$lua_dir/include/lua-5.2"
 			else
 				#
 				# The user didn't specify a directory in which liblua resides;
