--- CMakeLists.txt.orig	2026-02-24 08:07:25
+++ CMakeLists.txt	2026-02-25 07:51:23
@@ -46,8 +46,8 @@
                 DESTINATION "${GUI}.app/Contents/Resources/ghostscript")
 
         # ImageMagick
-        get_filename_component(IMAGEMAGICK_LIB_DIR
-                       ${ImageMagick_MagickCore_LIBRARY}
+        set(IMAGEMAGICK_LIB_DIR
+                       "%PREFIX%/lib/ImageMagick7/lib"
                        DIRECTORY)
 
         get_filename_component(IMAGEMAGICK_PREFIX
