--- cmake_aux/Modules/ARMA_FindSuperLU.cmake.orig	2015-11-09 22:42:43.000000000 -0600
+++ cmake_aux/Modules/ARMA_FindSuperLU.cmake	2015-12-26 10:21:57.000000000 -0600
@@ -12,11 +12,12 @@
   /usr/local/include/superlu/
   /usr/local/include/SuperLU/
   /usr/local/include/
+  /opt/local/include/
 )
 
 find_library(SuperLU_LIBRARY
   NAMES superlu
-  PATHS /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib
+  PATHS /usr/lib64 /usr/lib /usr/local/lib64 /usr/local/lib /opt/local/lib
 )
 
 SET(SuperLU_FOUND NO)
