--- src/luaconf.h.orig	2013-04-12 14:48:47.000000000 -0400
+++ src/luaconf.h	2015-02-20 07:03:28.000000000 -0500
@@ -100,7 +100,7 @@
 #else			/* }{ */
 
 #define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"@PREFIX@/"
 #define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
 #define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
