--- modules/gui/macosx/intf.m.orig	2016-02-11 13:33:24.000000000 -0600
+++ modules/gui/macosx/intf.m	2016-02-11 13:42:18.000000000 -0600
@@ -68,8 +68,6 @@
 #import "VideoEffects.h"
 #import "AudioEffects.h"
 
-#import <Sparkle/Sparkle.h>                 /* we're the update delegate */
-
 #import "iTunes.h"
 #import "Spotify.h"
 
@@ -785,8 +783,6 @@
     PL_LOCK;
     items_at_launch = p_playlist->p_local_category->i_children;
     PL_UNLOCK;
-
-    [[SUUpdater sharedUpdater] setDelegate:self];
 }
 
 - (void)applicationDidFinishLaunching:(NSNotification *)aNotification
@@ -974,25 +970,6 @@
 }
 
 #pragma mark -
-#pragma mark Sparkle delegate
-/* received directly before the update gets installed, so let's shut down a bit */
-- (void)updater:(SUUpdater *)updater willInstallUpdate:(SUAppcastItem *)update
-{
-    [NSApp activateIgnoringOtherApps:YES];
-    [o_remote stopListening: self];
-    [[VLCCoreInteraction sharedInstance] stop];
-}
-
-/* don't be enthusiastic about an update if we currently play a video */
-- (BOOL)updaterMayCheckForUpdates:(SUUpdater *)bundle
-{
-    if ([self activeVideoPlayback])
-        return NO;
-
-    return YES;
-}
-
-#pragma mark -
 #pragma mark Media Key support
 
 -(void)mediaKeyTap:(SPMediaKeyTap*)keyTap receivedMediaKeyEvent:(NSEvent*)event
