--- extras/package/macosx/Resources/English.lproj/Preferences.xib.orig	2016-02-11 13:33:22.000000000 -0600
+++ extras/package/macosx/Resources/English.lproj/Preferences.xib	2016-02-11 13:48:00.000000000 -0600
@@ -4779,9 +4779,6 @@
 				<string key="NSMaxSize">{10000000000000, 10000000000000}</string>
 				<bool key="NSWindowIsRestorable">YES</bool>
 			</object>
-			<object class="NSCustomObject" id="761483302">
-				<string key="NSClassName">SUUpdater</string>
-			</object>
 			<object class="NSWindowTemplate" id="898680423">
 				<int key="NSWindowStyleMask">15</int>
 				<int key="NSWindowBacking">2</int>
@@ -7629,22 +7626,6 @@
 					<int key="connectionID">2681</int>
 				</object>
 				<object class="IBConnectionRecord">
-					<object class="IBBindingConnection" key="connection">
-						<string key="label">value: automaticallyChecksForUpdates</string>
-						<reference key="source" ref="243384014"/>
-						<reference key="destination" ref="761483302"/>
-						<object class="NSNibBindingConnector" key="connector">
-							<reference key="NSSource" ref="243384014"/>
-							<reference key="NSDestination" ref="761483302"/>
-							<string key="NSLabel">value: automaticallyChecksForUpdates</string>
-							<string key="NSBinding">value</string>
-							<string key="NSKeyPath">automaticallyChecksForUpdates</string>
-							<int key="NSNibBindingConnectorVersion">2</int>
-						</object>
-					</object>
-					<int key="connectionID">3559</int>
-				</object>
-				<object class="IBConnectionRecord">
 					<object class="IBOutletConnection" key="connection">
 						<string key="label">nextKeyView</string>
 						<reference key="source" ref="588417130"/>
@@ -9360,12 +9341,6 @@
 						<reference key="parent" ref="492678965"/>
 					</object>
 					<object class="IBObjectRecord">
-						<int key="objectID">3555</int>
-						<reference key="object" ref="761483302"/>
-						<reference key="parent" ref="0"/>
-						<string key="objectName">Sparkle Updater</string>
-					</object>
-					<object class="IBObjectRecord">
 						<int key="objectID">3556</int>
 						<reference key="object" ref="243384014"/>
 						<array class="NSMutableArray" key="children">
