--- setup.py.orig	2020-05-30 18:41:13.000000000 -0500
+++ setup.py	2020-06-02 16:15:26.000000000 -0500
@@ -303,7 +303,7 @@
           author_email='tkeffer@gmail.com',
           url='http://www.weewx.com',
           license='GPLv3',
-          py_modules=['daemon', 'six'],
+          py_modules=['daemon'],
           package_dir={'': 'bin'},
           packages=['schemas',
                     'user',
