--- setup.cfg.orig	2022-10-24 15:14:21.000000000 -0500
+++ setup.cfg	2023-01-30 05:50:40.000000000 -0600
@@ -6,9 +6,9 @@
 # windows: c:\Users\weewx
 
 [install]
-home = /home/weewx
+#home = /home/weewx
 
-prefix =
-exec_prefix =
-install_lib = %(home)s/bin
-install_scripts = %(home)s/bin
+prefix = %%PREFIX%%
+exec_prefix = %(prefix)s/bin
+install_lib = %(prefix)s/bin
+install_scripts = %(prefix)s/bin
