--- setup.cfg.orig	2022-10-24 16:09:03.000000000 -0500
+++ setup.cfg	2022-10-24 18:29:49.000000000 -0500
@@ -1,9 +1,9 @@
 [install]
-home = /home/weewx
-prefix = 
-exec_prefix = 
-install_lib = %(home)s/bin
-install_scripts = %(home)s/bin
+#home = /home/weewx
+prefix = %%PREFIX%%
+exec_prefix = %(prefix)s/bin
+install_lib = %(prefix)s/bin
+install_scripts = %(prefix)s/bin
 
 [egg_info]
 tag_build = 
