--- Build.PL.orig	2017-12-27 04:03:49.000000000 -0600
+++ Build.PL	2019-02-16 11:29:28.000000000 -0600
@@ -99,7 +99,7 @@
     module_name       => 'App::ClusterSSH',
     license           => 'perl',
     dist_author       => q{Duncan Ferguson <duncan_j_ferguson@yahoo.co.uk>},
-    dist_version_from => 'lib/App/ClusterSSH.pm',
+    dist_version_from => 'lib/darwin-thread-multi-2level/App/ClusterSSH.pm',
     requires          => {
         'version'           => '0.77',
         'Tk'                => '800.022',
