--- lib/vtls/gtls.c.orig	2014-12-28 07:36:05.000000000 -0600
+++ lib/vtls/gtls.c	2015-02-24 09:59:28.000000000 -0600
@@ -618,7 +618,7 @@
       gnutls_alpn_set_protocols(session, protocols, protocols_size, 0);
       infof(data, "ALPN, offering %s, %s\n", NGHTTP2_PROTO_VERSION_ID,
             ALPN_HTTP_1_1);
-      connssl->asked_for_h2 = TRUE;
+      conn->ssl[sockindex].asked_for_h2 = TRUE;
     }
     else {
       infof(data, "SSL, can't negotiate HTTP/2.0 without ALPN\n");
@@ -1048,7 +1048,7 @@
         conn->negnpn = NPN_HTTP1_1;
       }
     }
-    else if(connssl->asked_for_h2) {
+    else if(conn->ssl[sockindex].asked_for_h2) {
       infof(data, "ALPN, server did not agree to a protocol\n");
     }
   }
