*** blockhosts.cfg.orig	2007-12-27 07:07:10.000000000 -0800
--- blockhosts.cfg	2007-12-27 10:08:00.000000000 -0800
***************
*** 84,88 ****
  # blockhosts section defines the log files to scan and patterns to look for
  
! #LOGFILES = [ "/var/log/secure", ] # default
  #LOGFILES = [ "/var/log/auth.log", ]
  #LOGFILES = [ "/var/log/secure", "/var/log/vsftpd.log", ]
--- 84,88 ----
  # blockhosts section defines the log files to scan and patterns to look for
  
! LOGFILES = [ "/var/log/system.log", ] # default
  #LOGFILES = [ "/var/log/auth.log", ]
  #LOGFILES = [ "/var/log/secure", "/var/log/vsftpd.log", ]
