--- Portfile.orig	2014-06-22 07:43:59.000000000 -0500
+++ Portfile	2014-06-22 10:28:55.000000000 -0500
@@ -8,6 +8,7 @@
 name                apache24-devel
 apache2.name        apache2
 version             2.4.9
+revision            1
 categories          www
 maintainers         ryandesign mathiesen.info:macintosh pixilla openmaintainer
 license             Apache-2
@@ -129,7 +130,14 @@
     }
 }
 
-# post-activate {
+ post-activate {
+    set confDir ${prefix}/etc/${apache2.name}
+    if {![file exists ${confDir}/httpd.conf]} {
+        file copy ${confDir}/httpd.conf.orig ${confDir}/httpd.conf
+        reinplace "s|#LoadModule slotmem_shm_module|LoadModule slotmem_shm_module|g" \
+        ${confDir}/httpd.conf
+    }
+
 # 
 #     if {![file exists ${apache2.sbindir}/envvars]} {
 #         xinstall -m 755 ${apache2.sbindir}/envvars-std ${apache2.sbindir}/envvars
@@ -148,7 +156,7 @@
 #             exec "${apache2.sbindir}/a2${action}" ${module}
 #         }
 #     }
-# }
+ }
 
 platform openbsd {
     build.env-append "LD_LIBRARY_PATH=${prefix}/lib"
