--- CMakeLists.txt.orig	2016-02-13 13:03:50.000000000 -0600
+++ CMakeLists.txt	2016-02-17 19:16:40.000000000 -0600
@@ -569,9 +569,9 @@
 	if(NOT MSVC AND NOT BORLAND)
 		add_subdirectory( utils )
 	endif(NOT MSVC AND NOT BORLAND)
-	if(NOT WIN32)
-		add_subdirectory( fonts )
-	endif(NOT WIN32)
+#	if(NOT WIN32)
+#		add_subdirectory( fonts )
+#	endif(NOT WIN32)
 #	add_subdirectory( mgllab )
 endif(NOT enable-lgpl)
 
