--- src/include/mpi.h.in.orig	2024-10-02 09:30:59
+++ src/include/mpi.h.in	2024-10-12 05:47:57
@@ -335,18 +335,6 @@
 #define MPI_WIN_DISP_UNIT     0x66000005
 #define MPI_WIN_CREATE_FLAVOR 0x66000007
 #define MPI_WIN_MODEL         0x66000009
-
-/* Definitions that are determined by configure. */
-typedef @MPI_AINT@ MPI_Aint;
-typedef @MPI_FINT@ MPI_Fint;
-typedef @MPI_COUNT@ MPI_Count;
-
-/* Let ROMIO know that MPI_Offset is already defined */
-#define HAVE_MPI_OFFSET
-/* MPI_OFFSET_TYPEDEF is set in configure and is
-      typedef $MPI_OFFSET MPI_Offset;
-   where $MPI_OFFSET is the correct C type */
-@MPI_OFFSET_TYPEDEF@
 
 /* The order of these elements must match that in mpif.h, mpi_f08_types.f90,
    and mpi_c_interface_types.f90 */
@@ -386,6 +374,18 @@
 /* Provided in libmpifort.so */
 extern MPIU_DLL_SPEC MPI_F08_status *MPI_F08_STATUS_IGNORE MPICH_API_PUBLIC;
 extern MPIU_DLL_SPEC MPI_F08_status *MPI_F08_STATUSES_IGNORE MPICH_API_PUBLIC;
+
+/* Definitions that are determined by configure. */
+typedef @MPI_AINT@ MPI_Aint;
+typedef @MPI_FINT@ MPI_Fint;
+typedef @MPI_COUNT@ MPI_Count;
+
+/* Let ROMIO know that MPI_Offset is already defined */
+#define HAVE_MPI_OFFSET
+/* MPI_OFFSET_TYPEDEF is set in configure and is
+      typedef $MPI_OFFSET MPI_Offset;
+   where $MPI_OFFSET is the correct C type */
+@MPI_OFFSET_TYPEDEF@
 
 #ifdef MPICH_DEFINE_ATTR_TYPE_TYPES
 static const MPI_Datatype mpich_mpi_datatype_null MPICH_ATTR_TYPE_TAG_MUST_BE_NULL() = MPI_DATATYPE_NULL;
