--- xeusConfig.cmake.in.orig	2020-06-02 01:43:40.000000000 -0500
+++ xeusConfig.cmake.in	2020-06-19 08:48:44.000000000 -0500
@@ -33,8 +33,8 @@
     # In order to not propagate the build-time requirement for pkg-config, we
     # do not propagate the dependency on ZeroMQ.
 
-    #find_package(PkgConfig)
-    #pkg_check_modules(ZeroMQ libzmq>=@zeromq_REQUIRED_VERSION@ REQUIRED)
+    find_package(PkgConfig)
+    pkg_check_modules(ZeroMQ libzmq>=@zeromq_REQUIRED_VERSION@ REQUIRED)
 endif()
 
 find_dependency(cppzmq @cppzmq_REQUIRED_VERSION@)
