--- src/Makefile.orig	2015-02-01 13:10:22.000000000 -0600
+++ src/Makefile	2015-02-01 13:56:28.000000000 -0600
@@ -9,7 +9,7 @@
 all: $(OBJ)
 
 mpfr_%.oct: mpfr_%.cc mpfr_commons.cc
-	$(MKOCTFILE) $(CFLAGS) -o $@ $(LDFLAGS) $(LDFLAGS_MPFR)  $<
+	$(MKOCTFILE) " " $(CFLAGS) -o $@ $(LDFLAGS_MPFR)  $<
 
 # The following build target are not required for the package installation
 # process, but are used during development of the package.
