--- src/fastlscomplex.cc.orig	2012-08-16 16:00:23.000000000 -0500
+++ src/fastlscomplex.cc	2018-06-15 09:34:37.000000000 -0500
@@ -59,11 +59,12 @@
       int noctaves = args(3).int_value ();
       int ncoeff = args(4).int_value ();
 
-      if (tvals.numel () != xvals.numel ())
+      if (tvals.numel () != xvals.numel ()) {
         if (tvals.numel () > xvals.numel ())
           error ("fastlscomplex: More time values than magnitude values");
         else
           error ("fastlscomplex: More magnitude values than time values");
+      }
       if (ncoeff == 0)
         error ("fastlscomplex: No coefficients to compute");
       if (noctaves == 0)
