--- libguile/stime.c.orig	2014-02-28 14:01:27.000000000 -0600
+++ libguile/stime.c	2016-06-22 08:40:26.000000000 -0500
@@ -832,19 +832,19 @@
   if (clock_gettime (CLOCK_REALTIME, &posix_real_time_base) == 0)
     get_internal_real_time = get_internal_real_time_posix_timer;
 
-#ifdef HAVE_POSIX_CPUTIME
+/* #ifdef HAVE_POSIX_CPUTIME
   {
     clockid_t dummy;
     
     /* Only use the _POSIX_CPUTIME clock if it's going to work across
-       CPUs. */
+       CPUs. *\/
     if (clock_getcpuclockid (0, &dummy) == 0 &&
         clock_gettime (CLOCK_PROCESS_CPUTIME_ID, &posix_run_time_base) == 0)
       get_internal_run_time = get_internal_run_time_posix_timer;
     else
       errno = 0;
   }
-#endif /* HAVE_POSIX_CPUTIME */
+#endif /* HAVE_POSIX_CPUTIME *\/ */
 #endif /* HAVE_CLOCKTIME */
 
   /* If needed, init and use gettimeofday timer. */
